From 833c6e2263e3ef3042a80c4da818be5ed3108c3a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Sun, 2 Aug 2015 22:16:23 +0200
Subject: [PATCH 07/11] backend-x11: Adapt commit 55fd05ea5 to the 3-16 branch

Still only compile-tested, but I was asked repeatedly to apply this
patch to 3-16, so let's try again ...
---
 src/backends/x11/meta-backend-x11.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/src/backends/x11/meta-backend-x11.c b/src/backends/x11/meta-backend-x11.c
index 6e52a5a..3ff8431 100644
--- a/src/backends/x11/meta-backend-x11.c
+++ b/src/backends/x11/meta-backend-x11.c
@@ -141,7 +141,15 @@ translate_crossing_event (MetaBackendX11 *x11,
       return;
     }
 
-  enter_event->event = meta_backend_x11_get_xwindow (x11);
+  Window stage_window = meta_backend_x11_get_xwindow (x11);
+  if (enter_event->event != stage_window)
+    {
+      /* See above for the rationale for this... */
+      g_assert (!meta_is_wayland_compositor ());
+      enter_event->event = meta_backend_x11_get_xwindow (x11);
+      enter_event->event_x = enter_event->root_x;
+      enter_event->event_y = enter_event->root_y;
+    }
 }
 
 static void
-- 
2.5.0

